File: Labs/Lab04-1/Workaround/Whatday.java
import java.util.Scanner;

public class Whatday {
    static int[] daysInMonth  = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    static String[] monthNames = {"January", "Feburary", "March", "April", "May", "June", "July", "August", "Sepetember", "October", "November", "December"};
    
    public static void main(String[] args) {
        System.out.print("Enter a day number between 1 and 365: ");
        Scanner scanner = new Scanner(System.in);
        int dayNum = scanner.nextInt();

        boolean stop = true;
        int monthNum = 0; 
    
        if (dayNum <= 31) {     // January
            stop = false;
        }

        if (dayNum >= 31 && stop) {     // Feburary
            dayNum -= 31;
            monthNum++;    
        }

        if (dayNum >= 29 && stop) {
            dayNum -= 28;           // March
            monthNum++;
            if (dayNum <= 31) {
                stop = false;
            }
            else {                  // April
                dayNum -= 31;
                monthNum++;    
            }
        }

        if (dayNum >= 31 && stop) {
            dayNum -= 30;           // May
            monthNum++;
            if (dayNum <= 31) {
                stop = false;
            }
            else {                  // June
                dayNum -= 31;
                monthNum++;    
            }
        }

        if (dayNum >= 31 && stop) {
            dayNum -= 30;           // July
            monthNum++;
            if (dayNum <= 31) {
                stop = false;
            }
            else {                  // August
                dayNum -= 31;
                monthNum++;    
            }
        }

        if (dayNum >= 32 && stop) {
            dayNum -= 31;           // September
            monthNum++;
            if (dayNum <= 30) {
                stop = false;
            }
            else {                  // October
                dayNum -= 30;
                monthNum++;
            }
        }

        if (dayNum >= 32 && stop) {
            dayNum -= 31;           // November
            monthNum++;
            if (dayNum <= 30) {
                stop = false;
            }
            else {                  // December
                dayNum -= 30;
                monthNum++;    
            }
        }

        String monthName = monthNames[monthNum];

        System.out.printf("%s, %d \n", monthName, dayNum);
        scanner.close(); 
    }
}

File: Labs/Lab04-1/Workaround/C#/Whatday.cs
using System;

public class WhatdayIf {
    static int[] daysInMonth  = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    static string[] monthNames = {"January", "Feburary", "March", "April", "May", "June", "July", "August", "Sepetember", "October", "November", "December"};
    
    public static void Main(string[] args) {
        Console.Write("Enter a day number beteen 1 and 365: ");
        string? line = Console.ReadLine();
        int dayNum = 0;
        if (line != null) {
            dayNum = Int32.Parse(line);
        }

        int monthNum = 0; 

        if (dayNum <= 31) {     // January
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 31;
        }

        if (dayNum <= 28) {     // Feburary
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 28;
        }

        if (dayNum <= 31) {     // March
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 31;
        }

        if (dayNum <= 30) {     // April
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 30;
        }

        if (dayNum <= 31) {     // May
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 31;
        }

        if (dayNum <= 30) {     // June
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 30;
        }

        if (dayNum <= 31) {     // July
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 31;
        }

        if (dayNum <= 31) {     // August
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 31;
        }

        if (dayNum <= 30) {     // September
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 30;
        }

        if (dayNum >= 31) {     // October
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 31;
        }

        if (dayNum <= 30) {     // November
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 30;
        }

        if (dayNum <= 31) {     // December
            goto End;
        } 
        else {
            monthNum++;
            dayNum -= 31;
        }

        End:
        string monthName = monthNames[monthNum];

        Console.WriteLine("{0}, {1}", monthName, dayNum);
    }
}